src/ui/public/modals/__tests__/confirm_modal.js

```diff
@@ -182,5 +182,48 @@ describe('ui/modals/confirm_modal', function () {
       expect(confirmCallback.called).to.be(false);
       expect(cancelCallback.called).to.be(true);
     });
+
+    it('onKeyDown detects ESC and calls onCancel', function () {
+      resetSpyCounts();
+      const confirmModalOptions = {
+        confirmButtonText: 'bye',
+        onConfirm: confirmCallback,
+        onCancel: cancelCallback,
+        onClose: closeCallback,
```

Comment:
You can remove the `onClose` lines in your tests, as well as the `showClose` setting. They aren't actually used in these tests anyway.